pymc/model.py

```diff
@@ -920,50 +928,228 @@ def test_point(self) -> Dict[str, np.ndarray]:
+        return_transformed: bool = False,
+    ) -> List[TensorVariable]:
         """Recomputes numeric initial values for all free model variables.
 
+        Parameters
+        ----------
+        jitter_rvs : set
+            The set (or list or tuple) of random variables for which a U(-1, +1) jitter should be
+            added to the initial value. Only available for variables that have a transform or real-valued support.
+        default_strategy : str
+            Which of { "moment", "prior" } to prefer if the initval setting for an RV is None.
+        overrides : dict
+            Initial value (strategies) to use instead of what's specified in `Model.initial_values`.
+        return_transformed : bool
+            Switches between returning the dictionary based on RV vars or RV value vars as keys.
+
+        Returns
+        -------
+        initial_points : list of TensorVariable
+            Aesara expressions for initial values of the free random variables.
+        """
+        from pymc.distributions.distribution import get_moment
+
+        if jitter_rvs is None:
+            jitter_rvs = set()
+
+        if overrides is None:
+            overrides = {}
+
+        initial_values = []
+        initial_values_transformed = []
+
+        for variable in self.free_RVs:
+            if variable in overrides:
+                strategy = overrides[variable]
+            else:
+                strategy = self.initial_values.get(variable, None)
+
+            if strategy is None:
+                strategy = default_strategy
+
+            if strategy == "moment":
+                value = get_moment(variable)
+            elif strategy == "prior":
+                value = variable
+            else:
+                value = at.as_tensor(strategy, dtype=variable.dtype).astype(variable.dtype)
+
+            transform = getattr(self.rvs_to_values[variable].tag, "transform", None)
+
+            if transform is not None:
+                value = transform.forward(variable, value)
+
+            if variable in jitter_rvs:
+                jitter = at.random.uniform(at.zeros_like(variable) - 1, at.zeros_like(variable) + 1)
+                jitter.name = "jitter"
+                value = value + jitter
+
+            initial_values_transformed.append(value)
+
+            if transform is not None:
+                value = transform.backward(variable, value)
+
+            initial_values.append(value)
+
+        # Copy the outputs in a way that we still know what corresponds to the free_RVS
+        all_inputs = []
+        for var in self.free_RVs:
+            all_inputs.extend(aesara.graph.graph_inputs([var]))
+
+        all_outputs = []
+        all_outputs.extend(self.free_RVs)
+        all_outputs.extend(initial_values)
+        all_outputs.extend(initial_values_transformed)
+
+        copy_graph = aesara.graph.FunctionGraph(outputs=all_outputs, clone=True)
```

Comment:
We should see if we can use the `memo` functionality as done in aeppl here: https://github.com/aesara-devs/aeppl/blob/d24eee80a7448c48b55a8ec41aec150d1dd9d6a7/aeppl/joint_logprob.py#L82-L103

Reply:
Not sure how we'd do that here; also not sure if it's worth the added complexity.
If you think it's worthwhile, can we defer that to a follow up?

Reply:
Yes, this is quite a hackish thing we did just to be able to get a reference between the old and new variables. Just because we didn't know about the memo thing

Reply:
I checked again, and the code does not change much if we use the `memo`, so not worth the bother.